src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -973,7 +982,13 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private async loadActiveContexts(kernelChangedArgs: nb.IKernelChangedArgs): Promise<void> {
 		if (kernelChangedArgs && kernelChangedArgs.newValue && kernelChangedArgs.newValue.name) {
 			let kernelDisplayName = this.getDisplayNameFromSpecName(kernelChangedArgs.newValue);
-			let context = NotebookContexts.getContextForKernel(this._activeConnection, this.getApplicableConnectionProviderIds(kernelDisplayName));
+			if (this.context?.serverCapabilities?.notebookKernelAlias && this.selectedKernelDisplayName === this.context?.serverCapabilities?.notebookKernelAlias) {
+				kernelDisplayName = this.context.serverCapabilities?.notebookKernelAlias;
+			}
+			let context;
```

Comment:
we should try to provide types when declaring variables.  If it really needs to be `any` then it should be explicitly declared that way.